engines/wintermute/base/gfx/opengl/base_surface_opengl3d.cpp

```diff
@@ -128,8 +128,27 @@ bool BaseSurfaceOpenGL3D::create(const Common::String &filename, bool defaultCK,
 	}
 
 	if (needsColorKey) {
-		Graphics::TransparentSurface trans(*_surf);
-		trans.applyColorKey(_ckRed, _ckGreen, _ckBlue, replaceAlpha);
+		// this is taken from Graphics::TransparentSurface
```

Comment:
Consider putting this functionality in a separate function.

Reply:
Done